🐛 Beta Bugfixes and Enhancements

hjamet opened the pull request:
# 🐛 Beta Bugfixes and Enhancements

This pull request mainly addresses the following issues:
- [x] https://github.com/mysticcoders/voicenotes-sync/issues/41
- [x] https://github.com/mysticcoders/voicenotes-sync/issues/40
- [x] https://github.com/mysticcoders/voicenotes-sync/issues/30

## 🦦 New Features and Improvements

### 1. 📝 Enhanced Custom Template Settings
- 🔧 Larger, auto-resizing input field for custom templates
- ✅ New checkbox option to use custom frontmatter
- 🆕 Addition of {{ parent_note }} variable in the custom note template

### 2. 📅 Improved Date Formatting
- 🎯 Default frontmatter now respects user-defined date format settings

### 3. 📊 Better Markdown Support
- 🧼 Cleaner synchronization of notes with proper Markdown handling
- 🚫 Elimination of random extra spaces

### 4. 🧹 Tidy Transcript Integration
- 🔄 Proper integration of Tidy Transcript when generated

### 5. 🛠️ Bug Fixes
- 🔒 Resolution of synchronization bug that could overwrite files
- 🗑️ Removal of unnecessary debug and sync time settings
- 🆕 Fixed subnotes synchronization bug: subnotes are now synchronized even if the parent note has already been synced

### 6. ⏱️ Sync Interval Validation
- 🔢 New safety check ensures sync interval is a number ≥ 1 minute

## 💖 Thank You Note

I'm thrilled to contribute to this fantastic plugin! Hope these updates bring joy to all Voicenotes users! 🎉🎊

kinabalu commented:
@hjamet thanks again for the work, let me know when this is ready for review

hjamet commented:
> @hjamet thanks again for the work, let me know when this is ready for review

Sorry : I missed the message! It's my pleasure and I think it should be ready for review ;)

kinabalu reviewed (commented):
generally looks good -- i'd like to have discussions about the comments made before we move to approve. thanks!

hjamet commented on main.ts:
```diff
@@ -112,11 +135,11 @@ export default class VoiceNotesPlugin extends Plugin {
     }
 
     async onload() {
-        window.clearInterval(this.syncInterval);
```
No: the problem arose from a duplication of the option specifying the interval between updates in the previous version. The code was therefore implemented twice in two different ways. I've removed the second way, leaving only the `syncTimeout` parameter, which correctly initialises the counter in the `setupAutoSync()` function :)

hjamet commented on main.ts:
```diff
@@ -210,129 +253,151 @@ export default class VoiceNotesPlugin extends Plugin {
-        const creations = Object.fromEntries(
-            creationTypes.map(type => [type, recording.creations.find((creation: { type: string }) => creation.type === type)])
-        );
+        // Vérifier si la note existe déjà
```
My bad, I'm correcting that right now!

hjamet commented on main.ts:
```diff
@@ -210,129 +253,151 @@ export default class VoiceNotesPlugin extends Plugin {
+                    const updatedFrontmatter = existingFrontmatter.replace(/recording_id:.*\n?/, '') + `recording_id: ${recording.recording_id}\n`;
+                    note = note.replace(frontmatterRegex, `---\n${updatedFrontmatter}---`);
+                } else {
+                    // No frontmatter, create one with recording_id
```
At https://github.com/mysticcoders/voicenotes-sync/issues/40's request, I've added an option allowing you to remove the default frontmatter and put your own in the Jinja2 template. However, during synchronisation, the recording_id parameter is essential, hence this addition after checking.

hjamet commented on settings.ts:
```diff
@@ -195,7 +205,7 @@ export class VoiceNotesSettingTab extends PluginSettingTab {
       .addText((text) =>
         text
-          .setPlaceholder('YYYY-MM-DD')
+          .setPlaceholder('DD/MM/YYYY HH:mm')
```
This was to show users that it was also possible to add the time, but if you prefer the original format, I'll leave it as is :)

hjamet commented on settings.ts:
```diff
@@ -216,44 +226,31 @@ export class VoiceNotesSettingTab extends PluginSettingTab {
+      .addToggle((toggle) => {
+        toggle
+          .setValue(this.plugin.settings.useDefaultFrontmatter)
+          .onChange(async (value) => {
```
Again, the purpose of this change is to respond to https://github.com/mysticcoders/voicenotes-sync/issues/40:
- If the option is set to True, the frontmatter used is the default frontmatter, with recording_id, duration, created_at and updated_at.
- Otherwise, the frontmatter does not exist (the user is therefore free to customise it in their jinja2 template). Only the recording_id property will be added for technical reasons.

_It is also possible to include the default frontmatter in the basic Jinja2 template, but I find that less clear and less practical than simply clicking on the option ! 🦦 _

hjamet commented:
I've also just added the following two changes to the update:

1. Removal of the "Excluded Folders" option, which wasn't implemented or applicable, in favour of the "Excluded Tags" option, which prevents notes with specific tags from being synchronised;
2. Transformation of "Manual Sync"/"Manual Sync overwrite" into "Manual Sync (quick)"/"Manual Sync (complete)". This change was necessary because, since the first Beta, Manual Sync had been causing bugs. Now, to perform 'Manual Sync overwrite', all you have to do is delete all your notes and then do 'Manual Sync (complete)'.

I've also fixed a small error concerning sub-notes, which weren't synchronised correctly if they were added after the parent note had already been synchronised. 🦦
